src/lib/device_tree/src/lib.rs

```diff
@@ -38,27 +69,41 @@ enum Token {
     End = 0x9,
 }
 
+/// Represents a path to device tree node.
+/// It is also being used by properties, in which case the last component of the path stores the
+/// name of the property.
 pub struct Path {
     depth: usize,
     len: [usize; MAX_DEPTH],
     buf: [[u8; MAX_NAME_SIZE]; MAX_DEPTH],
 }
 
 impl Path {
+    /// Returns the number number of components in the path.
```

Comment:
number number -> number

Reply:
I am very sorry, somehow changes in this PR got some unrelated changes in this file, reverted all of them to prevent the conflict with the other PR.